massa-protocol-worker-2/src/handlers/endorsement_handler/propagation.rs

```diff
@@ -1,80 +1,155 @@
+                                            .get_endorsement_refs()
+                                            .iter()
+                                            .filter_map(|id| {
+                                                if endorsement_ids.contains(id) {
+                                                    return None;
+                                                }
+                                                Some((
+                                                    *id,
+                                                    endorsements_reader.get(id).cloned().unwrap(),
+                                                ))
+                                            })
+                                            .collect()
+                                    };
+                                    for endorsement_id in new_endorsements.keys().copied() {
+                                        endorsement_ids.put(endorsement_id, ());
+                                    }
+                                    let to_send =
+                                        new_endorsements.into_values().collect::<Vec<_>>();
+                                    if !to_send.is_empty() {
+                                        let active_connections_read =
+                                            self.active_connections.read();
+                                        if let Some(peer) =
+                                            active_connections_read.connections.get(peer_id)
+                                        {
+                                            if let Err(err) = peer.send_channels.send(
+                                                &self.endorsement_serializer,
+                                                EndorsementMessage::Endorsements(to_send).into(),
+                                                false,
+                                            ) {
+                                                warn!("could not send endorsements batch to node {}: {}", peer_id, err);
+                                            }
+                                        }
+                                    }
                                 }
-
-                                // Send the endorsements
-                                let message = EndorsementMessage::Endorsements(endorsements);
-                                println!("Sending message to {:?}", peer_id);
-                                // TODO: Error management
-                                connection
-                                    .send_channels
-                                    .send(&endorsement_serializer, message.into(), false)
-                                    .unwrap();
                             }
                         }
                     }
                 }
                 Err(err) => {
-                    println!("Error: {:?}", err);
+                    println!("Err: {:#?}", err);
```

Comment:
maybe remove println! and use info!, warn!, error!

Reply:
Done !